src/coreneuron/io/reports/nrnreport.hpp

```diff
@@ -69,19 +121,67 @@ enum class TargetType {
     SectionApicalAll = 9,
 };
 
-// enumerate that defines the type of target report requested
-enum ReportType {
-    SomaReport,
-    CompartmentReport,
-    SynapseReport,
-    IMembraneReport,
-    SectionReport,
-    SummationReport,
-    LFPReport
-};
-
-// enumerate that defines the section type for a Section report
-enum SectionType { Cell, Soma, Axon, Dendrite, Apical, All };
+constexpr std::array<std::pair<TargetType, std::string_view>, 10> target_type_map{
+    {{TargetType::Compartment, "Compartment"},
+     {TargetType::Cell, "Cell"},
+     {TargetType::SectionSoma, "SectionSoma"},
+     {TargetType::SectionAxon, "SectionAxon"},
+     {TargetType::SectionDendrite, "SectionDendrite"},
+     {TargetType::SectionApical, "SectionApical"},
+     {TargetType::SectionSomaAll, "SectionSomaAll"},
+     {TargetType::SectionAxonAll, "SectionAxonAll"},
+     {TargetType::SectionDendriteAll, "SectionDendriteAll"},
+     {TargetType::SectionApicalAll, "SectionApicalAll"}}};
+
+inline std::string to_string(TargetType t) {
+    return to_string(t, target_type_map, "TargetType");
+}
+inline TargetType target_type_from_string(const std::string& s) {
```

Comment:
Do we use this elsewhere (my `git grep` isn't showing anything)? Also, I feel like having so many versions of `*_from_string` defeats the purpose of the template.

Reply:
no, this one in particular is not used in this version. The whole enum will be removed in my future PR. For this reason I did not go ultra-precise on this.

I do not see the problem of having many `*_from_string`. To me it brings a shiver of order in a world of chaos. Everything in coreneuron has an `ad-hoc` solution and we need to understand each one. Here we have the same behavior for all the enums. They are more handy and versatile IMHO. Why not like `to_sting`? Because `to_string` can discriminate given the input. The `*_from_string` does not have this luxury. I still find this the simplest solution.